src/audio/tdfb/tdfb.c

```diff
@@ -761,24 +769,43 @@ static int tdfb_prepare(struct comp_dev *dev)
 
 	/* Initialize filter */
 	cd->config = comp_get_data_blob(cd->model_handler, NULL, NULL);
-	if (cd->config) {
-		ret = tdfb_setup(cd, sourceb->stream.channels, sinkb->stream.channels);
-		if (ret < 0) {
-			comp_err(dev, "tdfb_prepare() error: tdfb_setup failed.");
-			goto err;
-		}
+	if (!cd->config)
+		goto err;
 
-		/* Clear in/out buffers */
-		memset(cd->in, 0, TDFB_IN_BUF_LENGTH * sizeof(int32_t));
-		memset(cd->out, 0, TDFB_IN_BUF_LENGTH * sizeof(int32_t));
+	ret = tdfb_setup(cd, sourceb->stream.channels, sinkb->stream.channels);
+	if (ret < 0) {
+		comp_err(dev, "tdfb_prepare() error: tdfb_setup failed.");
+		goto err;
+	}
 
-		ret = set_func(dev);
-		return ret;
+	/* Clear in/out buffers */
+	memset(cd->in, 0, TDFB_IN_BUF_LENGTH * sizeof(int32_t));
+	memset(cd->out, 0, TDFB_IN_BUF_LENGTH * sizeof(int32_t));
+
+	ret = set_func(dev);
+	if (ret)
+		goto err;
```

Comment:
by hardcoding your return value after err: this ret gets thrown away, it is better to just manually set ret above and not hardcode it

Reply:
Done